src/shared/particle_dynamics/fluid_dynamics/fluid_dynamics_complex.h

```diff
@@ -137,6 +141,30 @@ class BaseDensitySummationComplex
     }
 };
 
+class DensitySummationComplexKernel : public BaseDensitySummationComplexKernel, public DensitySummationInnerKernel {
+  public:
+    DensitySummationComplexKernel(const BaseDensitySummationComplexKernel& complexKernel,
+                                  const DensitySummationInnerKernel& innerKernel)
+        : BaseDensitySummationComplexKernel(complexKernel), DensitySummationInnerKernel(innerKernel) {}
+
+    template<class RealT, class InnerInteractionFunc, class ContactSummationFunc>
+    static void interaction(size_t index_i, Real dt, RealT* rho_sum, RealT rho0, RealT inv_sigma0, const RealT* mass,
+                            InnerInteractionFunc&& innerInteraction, ContactSummationFunc&& ContactSummation)
+    {
+        innerInteraction(index_i, dt);
+        RealT sigma = ContactSummation(index_i);
```

Comment:
"RealT" seems not good name. "RealType"?

Reply:
Ok, I'll also change `Vec` to `VecType`, to keep the same nomenclature.